dnainator-javafx/src/test/java/nl/tudelft/dnainator/javafx/drawables/phylogeny/LeafNodeTest.java

```diff
@@ -0,0 +1,59 @@
+package nl.tudelft.dnainator.javafx.drawables.phylogeny;
+
+import static org.junit.Assert.assertEquals;
+import javafx.stage.Stage;
+import nl.tudelft.dnainator.javafx.AllColorsInUseException;
+import nl.tudelft.dnainator.javafx.ColorServer;
+import nl.tudelft.dnainator.tree.TreeNode;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.testfx.framework.junit.ApplicationTest;
+
+/**
+ * Test the {@link LeafNode} of the phylogenetic tree.
+ */
+public class LeafNodeTest extends ApplicationTest {
+
+	private LeafNode ln;
+	private TreeNode tn;
+	private ColorServer cs;
+	private ColorServer csSpy;
+	
+	@Override
+	public void start(Stage stage) throws Exception {
+	}
+	
+	/**
+	 * Creates new database service.
+	 */
+	@Before
+	public void setup() {
+		MockitoAnnotations.initMocks(this);
+		tn = new TreeNode(null);
+		cs = new ColorServer();
+		csSpy = Mockito.spy(cs);
+		ln = new LeafNode(tn, csSpy);
+	}
+	
+	/**
+	 * Test the creation of a {@link LeafNode}.
+	 */
+	@Test
+	public void testCreate() {
+		assertEquals(2, ln.getChildren().size());
+	}
+	
+	/**
+	 * Test what happens when a {@link LeafNode} is clicked with the mouse.
+	 * @throws AllColorsInUseException when all colors are in use.
+	 */
+	@Test
+	public void testOnMouseClicked() throws AllColorsInUseException {
+		Mockito.doNothing().when(csSpy).askColor(Mockito.anyString());
```

Comment:
I'll leave it in here, putting it in setup would cause it having to throw/catch the exception.